lib/Conversion/TorchToTMTensor/TorchToTMTensor.cpp

```diff
@@ -1297,6 +1355,95 @@ class ConvertAtenScatterReduceTwoOp
+    Value inputTensor = adaptor.getSelf();
+    auto inputType = inputTensor.getType().cast<RankedTensorType>();
+    unsigned inputRank = inputType.getRank();
+
+    // Step 2. Fetch dimension to perform sort in.
+    int64_t dim;
+    if (!matchPattern(op.getDim(), m_TorchConstantInt(&dim)))
+      return rewriter.notifyMatchFailure(
+          op, "unimplemented: only constant dim value is supported");
+    dim = toPositiveDim(dim, inputRank);
+    if (!isValidDim(dim, inputRank)) {
+      return rewriter.notifyMatchFailure(op, "dim is statically invalid");
+    }
+
+    // Step 3. Fetch the order of sorting.
+    bool descending;
+    if (!matchPattern(op.getDescending(), m_TorchConstantBool(&descending)))
+      return rewriter.notifyMatchFailure(
+          op, "unimplemented: only constant descending value is supported");
+
+    // Step 4. Form a RankedTensorType with same shape as that of the input's
+    //         but with elemental type i64.
+    RankedTensorType indicesType =
+        RankedTensorType::get(inputType.getShape(), rewriter.getI64Type());
+
+    // Step 5. Generate indices tensor.
+    SmallVector<Value> dynDims;
+    for (unsigned i = 0; i < inputType.getRank(); i++) {
+      if (inputType.isDynamicDim(i)) {
+        dynDims.push_back(rewriter.create<tensor::DimOp>(loc, inputTensor, i));
+      }
+    }
+    Value initEmptyTensor = rewriter.create<tensor::EmptyOp>(
+        loc, inputType.getShape(), rewriter.getI64Type(), dynDims);
+    Value zeroConst = rewriter.create<arith::ConstantOp>(
+        loc, rewriter.getZeroAttr(rewriter.getI64Type()));
+    Value outVector =
+        rewriter.create<linalg::FillOp>(loc, zeroConst, initEmptyTensor)
+            .getResult(0);
+
+    SmallVector<AffineMap> indexingMaps = {
+        AffineMap::getMultiDimIdentityMap(inputRank, op.getContext())};
+    SmallVector<utils::IteratorType> iteratorTypes(
+        inputRank, utils::IteratorType::parallel);
+    Value indicesTensor =
+        rewriter
+            .create<linalg::GenericOp>(
+                loc, outVector.getType(), ValueRange{}, outVector, indexingMaps,
+                iteratorTypes,
+                [&](OpBuilder &b, Location loc, ValueRange args) {
+                  Value index = b.create<linalg::IndexOp>(loc, dim);
+                  index = castIndexToInt64(b, loc, index);
+                  b.create<linalg::YieldOp>(loc, index);
+                })
+            .getResult(0);
+
+    // Step 6. Create TMTensor::SortOp.
+    SmallVector<Value> operands;
+    operands.push_back(inputTensor);
+    operands.push_back(indicesTensor);
+    SmallVector<Type> elementTypes;
+    elementTypes.push_back(inputType.getElementType());
+    elementTypes.push_back(indicesType.getElementType());
+
+    TMTensor::SortOp sortOp = NULL;
+    LogicalResult logicalResult =
+        createTMTensorSortOp(rewriter, loc, operands, elementTypes,
+                             /*dimension=*/dim, /*isStable=*/true,
```

Comment:
The documentation says that the default value for `isStable` is `False`: https://pytorch.org/docs/stable/generated/torch.sort.html

Can you add the last example in the documentation as an e2e test to make sure we are setting the right default value here?

Reply:
That's correct but the `aten.sort` op's definition generated within `GeneratedTorchOps.td` doesn't seem to bear the `isStable` argument.

Let me know what needs to be done here. I used the `JITOperatorRegistryDump.txt` to find the op's definition.

Reply:
Yeah, the op you're adding does not take `stable` as an argument and uses the default value. If you do specify `stable` in PyTorch, the op generated is `aten.sort.stable` rather than `aten.sort`. Given that the default value of PyTorch is `stable=False`, we should set it to `false` here too (I would add a comment mentioning that this is the default value and link to the docs).

For example, when I run:

```python
In [8]: x = torch.tensor([0, 1] * 9)

In [9]: torch.ops.aten.sort(x)
Out[9]:
(tensor([0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1]),
 tensor([ 2, 16,  4,  6, 14,  8,  0, 10, 12,  9, 17, 15, 13, 11,  7,  5,  3,  1]))
```

I get the unstable result.